chain/client/src/stateless_validation/endorsement_tracker.rs

```diff
@@ -0,0 +1,141 @@
+use std::collections::HashMap;
+use std::sync::Arc;
+
+use near_chain_primitives::Error;
+use near_epoch_manager::EpochManagerAdapter;
+use near_primitives::block_body::ChunkEndorsementSignatures;
+use near_primitives::checked_feature;
+use near_primitives::sharding::{ChunkHash, ShardChunkHeader};
+use near_primitives::stateless_validation::ChunkEndorsement;
+use near_primitives::types::AccountId;
+
+use crate::Client;
+
+// This is the number of unique chunks for which we would track the chunk endorsements.
+// Ideally, we should not be processing more than num_shards chunks at a time.
+const NUM_CHUNK_ENDORSEMENTS_CACHE_COUNT: usize = 100;
```

Comment:
nit: maybe call it NUM_CHUNKS_IN_CHUNK_ENDORSEMENTS_CACHE? It's more descriptive.